pkg/bindings/images/images.go

```diff
@@ -0,0 +1,189 @@
+	if err != nil {
+		return false, err
+	}
+	response, err := conn.NewRequest(http.MethodGet, fmt.Sprintf("/images/%s/exists", nameOrID), nil, nil)
+	if err != nil {
+		return false, err
+	}
+	if response.StatusCode == http.StatusOK {
+		return true, nil
+	}
+	return false, nil
+}
+
+// List returns a list of images in local storage.  The all boolean and filters parameters are optional
+// ways to alter the image query.
+func List(ctx context.Context, all *bool, filters map[string][]string) ([]handlers.ImageSummary, error) {
+	var imageSummary []handlers.ImageSummary
+	conn, err := bindings.GetConnectionFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	params := make(map[string]string)
+	if all != nil {
+		params["all"] = strconv.FormatBool(*all)
+	}
+	if filters != nil {
+		strFilters, err := bindings.FiltersToString(filters)
+		if err != nil {
+			return nil, err
+		}
+		params["filters"] = strFilters
+	}
+	response, err := conn.NewRequest(http.MethodGet, "/images/json", nil, params)
+	if err != nil {
+		return imageSummary, err
+	}
+	return imageSummary, response.Process(&imageSummary)
+}
+
+// Get performs an image inspect.  To have the on-disk size of the image calculated, you can
+// use the optional size parameter.
+func GetImage(ctx context.Context, nameOrID string, size *bool) (*inspect.ImageData, error) {
+	conn, err := bindings.GetConnectionFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	params := make(map[string]string)
+	if size != nil {
+		params["size"] = strconv.FormatBool(*size)
+	}
+	inspectedData := inspect.ImageData{}
+	response, err := conn.NewRequest(http.MethodGet, fmt.Sprintf("/images/%s/json", nameOrID), nil, params)
+	if err != nil {
+		return &inspectedData, err
+	}
+	return &inspectedData, response.Process(&inspectedData)
+}
+
+func ImageTree(ctx context.Context, nameOrId string) error {
+	return bindings.ErrNotImplemented
+}
+
+// History returns the parent layers of an image.
+func History(ctx context.Context, nameOrID string) ([]handlers.HistoryResponse, error) {
+	var history []handlers.HistoryResponse
+	conn, err := bindings.GetConnectionFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	response, err := conn.NewRequest(http.MethodGet, fmt.Sprintf("/images/%s/history", nameOrID), nil, nil)
+	if err != nil {
+		return history, err
+	}
+	return history, response.Process(&history)
+}
+
+func Load(ctx context.Context, r io.Reader) error {
+	conn, err := bindings.GetConnectionFromContext(ctx)
+	if err != nil {
+		return err
+	}
+	// TODO this still needs error handling added
+	//_, err := http.Post(c.makeEndpoint("/images/loads"), "application/json", r) //nolint
+	_ = conn
+	return bindings.ErrNotImplemented
+}
+
+// Remove deletes an image from local storage.  The optional force parameter will forcibly remove
+// the image by removing all all containers, including those that are Running, first.
+func Remove(ctx context.Context, nameOrID string, force *bool) ([]map[string]string, error) {
```

Comment:
Given this is one image, an array of maps seems wrong.

Reply:
discussed, keep for now